dnf/base.py

```diff
@@ -1381,8 +1381,15 @@ def cond_check(pkg):
                 if (pkg.basearchonly):
                     query_args.update({'arch': basearch})
                 q = self.sack.query().filter(**query_args).apply()
-                if not q or not cond_check(pkg):
-                    # a conditional package with unsatisfied requiremensts
+                if not q:
+                    package_string = pkg.name
+                    if pkg.basearchonly:
+                        package_string = package_string + '.' + basearch
```

Comment:
I would change it to either: `package_string = "{}.{}".format(package_string, basearch)` or `package_string += "." + basearch`.